### ISSUE TITLE
only publish pre-trained model's checkpoint files are useless.

### ISSUE THREAD

**Issue by civilman628** [tensorflow/tensorflow#6954]
@nathansilberman @sguada
I can see there are some pre-trained models in the link below for download.

https://github.com/tensorflow/models/tree/master/slim

However, these are only checkpoint files. We can't use them at all. Because without .pbtxt file, which is the graph define, we can't restore the any model only base on .ckpt file. There are no example about how to use these checkpoint file, such as for **a single image prediction** or feature extraction. I can only see one example about how to use V3 .pb file for prediction, but that one is .pb file, not .ckpt file. Without .pbtxt file, we can't freeze the graph at all.

I want to use Inception V2 model, which is smaller than V3 and better than V1. However, I can do nothing...

even if you fine tune the model, you can only evaluate it base on a batch file, but not to predict single image. in the section of **Evaluating performance of a model** of the page, we can only get accuracy. However, we need to test single image to see the result, but not a accuracy number for a batch file.

**Comment by ppries** [comment]
The graph definitions reside in https://github.com/tensorflow/models/tree/master/slim/nets. You should be able to create the graphs and load the appropriate checkpoint file using the code in that folder.

**Comment by civilman628** [comment]
While .py is different from .pbtxt. These pre-trained models use Imagenet dataset, which has 1000 classes and 1.2 Million images and more than 100+GB in size.

If I want ONLY want to get .pbtxt file for each model ( since .ckpt file is already there),  I need download that 1.2 million images and create data records to train at my local machine? 

To get a text graph define .pbtxt file for the model need download 100+GB file?

No one find this issue before?

**Comment by inflation** [comment]
@civilman628 You don't need to have the training dataset to get the structure of a graph. They are separated by design.

**Comment by civilman628** [comment]
I did not find any way to get model's .pbtxt file only from .py file. If we do not have .pbtxt, these pre-trained .ckpt can do nothing. We can't use .ckpt file for prediction or feature extraction.

**Comment by yaroslavvb** [comment]
@civilman628 this sounds like it could be a documentation issue. Can you specify which model you are trying to restore, and where you expected to see instructions on restoring it?

`.ckpt` files are indeed insufficient on their own, usually you need a  `.py` file and instructions that tells you how to run it

**Comment by civilman628** [comment]
any model. base on this link:

https://www.tensorflow.org/versions/master/tutorials/image_recognition/   it is for V3 only.

You can only predict an image base .pb file, This .pb file is from .ckpt and .pbtxt, by freeze graph right? If we only have pre-trained model .ckpt file. we can do nothing.

Furthermore, we can only get tensor name from .pb file, but not .ckpt file. Why there are so many different formats and there is no easy way to convert?

**Comment by civilman628** [comment]
There is no example about how to use .ckpt file for prediction and feature extraction. if we can't do prediction or feature extraction, what these .ckpt files use for?

**Comment by civilman628** [comment]
if no one report this issue before, then that's mean no one use these .ckpt files, since if anyone want to use them, then they will meet the same problem as mine.

**Comment by inflation** [comment]
Quick example:
`import tensorflow as tf
from nets import lenet

x = tf.placeholder(shape=[64,64,64,3], dtype=tf.float32)
lenet = lenet.lenet(x)

sess = tf.Session()
tf.train.write_graph(sess.graph_def, '/tmp/my-model', 'train.pbtxt')`

**Comment by yaroslavvb** [comment]
@civilman628 I'm guessing those `.ckpt` files are read by some `.py` script that sets up the model, and loads the checkpoint. They are not meant to be used directly.

**Comment by yaroslavvb** [comment]
OK, I'm assuming that's working as intended, `.ckpt` files are useless on their own, but are intended to be used by a script that builds a model, sets up saver, and loads them. This seems like a request for more documentation, but it's missing details of which/where documentation is missing. I'll close it for now, but if you can provide more information I'll reopen

**Comment by civilman628** [comment]
the above approach, which get .pbtxt and then freeze into .pb does not work, see the defect:

#7172

**Comment by parthg** [comment]
Actually, the .ckpt files are sufficient with the model code. Fore details, please see my answer here #7172

**Comment by dmoham1476** [comment]
vgg_19 and inception .pb files that we get from the pretrained models link are not frozen graphs. Without having a checkpoint, I cannot use freeze_graph.py. How can I generate a frozen graph from the pretrained model here? It would have been nice if the pretrained models link provided the .pb file along with the checkpoint, so we could freeze it easily.

**Comment by smohan10** [comment]
I am trying to use a pretrained nasnet from the tf slim and I see only model.ckpt.data-00000-of-00001 and model.ckpt.index. I am unsure if I can do anything with just these 2 files without the meta. Any way to convert it to .pb ? Also wondering why would tensorflow not release a .meta file!

**Comment by Qasim-Abdul-Majeed** [comment]
Hi, You can download its weights from here:
https://github.com/fchollet/deep-learning-models/releases/download/v0.5/inception_v3_weights_tf_dim_ordering_tf_kernels.h5

After downloading this, you can get values of its convolution  filters in a sequence using 'h5py'
e.g.
h5 = h5py('Path/to/.h5')
conv_name = 'conv2d_1'
conv2d_1 = h5[conv_name][conv_name]['kernel:0']

Now, conv2d_1 contains filter values of its first convolution filter in a model.
